spec/controllers/admin_controller_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe AdminController, type: :controller do
+  describe '#authenticate_admin' do
```

Comment:
I believe that the controllers that are actually executing this path should cover these tests already. @brendandeere what do you think about this testing a private method?